fix crash when enter btn is clicked and seed suggestion is to zero

Sirmorrison opened the pull request:
Fix #466

Sirmorrison committed:
fix crash when enter btn is clicked and seed suggestion is to zero

beansgum reviewed (changes requested):
Crashed when I tapped enter. The seed suggestion was visible but no item was focused.

```
panic: runtime error: index out of range [2] with length 1

goroutine 16 [running]:
github.com/planetdecred/godcr/ui.(*createRestore).handle(0xc000b8a000)
        /Users/collins/go/src/github.com/planetdecred/godcr/ui/create_restore_page.go:868 +0x140f
github.com/planetdecred/godcr/ui.(*mainPage).Layout(0xc000cc6a00, 0x0, 0x0, 0x320, 0x258, 0x3f8000003f800000, 0x5102c40, 0xc0001ebc90, 0xc02b0d6237b39908, 0x9262235f2, ...)
        /Users/collins/go/src/github.com/planetdecred/godcr/ui/main_page.go:234 +0x102
github.com/planetdecred/godcr/ui.(*Window).layoutPage.func2(0x0, 0x0, 0x320, 0x258, 0x3f8000003f800000, 0x5102c40, 0xc0001ebc90, 0xc02b0d6237b39908, 0x9262235f2, 0x59898a0, ...)
        /Users/collins/go/src/github.com/planetdecred/godcr/ui/window.go:133 +0x95
gioui.org/layout.Stack.Layout(0x4cc7301, 0x320, 0x258, 0x320, 0x258, 0x3f8000003f800000, 0x5102c40, 0xc0001ebc90, 0xc02b0d6237b39908, 0x9262235f2, ...)
        /Users/collins/go/pkg/mod/gioui.org@v0.0.0-20210418151603-3b69b5ed0512/layout/stack.go:60 +0x7d5
github.com/planetdecred/godcr/ui.(*Window).layoutPage(0xc00018c000, 0x320, 0x258, 0x320, 0x258, 0x3f8000003f800000, 0x5102c40, 0xc0001ebc90, 0xc02b0d6237b39908, 0x9262235f2, ...)
        /Users/collins/go/src/github.com/planetdecred/godcr/ui/window.go:127 +0x257
github.com/planetdecred/godcr/ui.(*Window).Loop(0xc00018c000, 0xc0001ebc00, 0xc0007910e0)
        /Users/collins/go/src/github.com/planetdecred/godcr/ui/window.go:235 +0x409
created by main.main
        /Users/collins/go/src/github.com/planetdecred/godcr/main.go:91 +0x525
```

Sirmorrison committed:
fix enter btn click bug on create restore page
- fix crash when a seed has been selected and the editor is focused
again,if enter btn is clicked, the app crash
- fix crash when the navigation key is used to select a seed option,
and the focus is moved to another editor, and the enter ban is pressed.
- fix bug with suggestion menu pop up hiding and not opening
when focus leaves the app and returns
